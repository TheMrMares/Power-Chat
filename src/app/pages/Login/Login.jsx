import React from 'react'
import PropTypes from 'prop-types'
import {
  Paper,
  withStyles,
} from '@material-ui/core'
import styles from './Login.style'
import abc from './Login.module.scss'

export class Login extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
  }

  render() {
    const { classes } = this.props
    return (
      <div className={classes.root}>
        <Paper className={classes.paper}>
          login
        </Paper>
        <p className={abc.testClass}>XDD</p>
      </div>
    )
  }
}

export default withStyles(styles)(Login)
