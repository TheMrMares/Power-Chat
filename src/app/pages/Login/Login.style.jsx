export default theme => ({ // eslint-disable-line
  paper: {
    padding: theme.spacing.unit * 2,
  },
})
