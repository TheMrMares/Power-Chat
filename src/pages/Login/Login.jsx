import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import Button from '../../components/Button'
import firebase from '../../firebase'
import styles from './Login.module.scss'
import Input from '../../components/Input'

export class Login extends React.Component {
  static propTypes = {
    history: PropTypes.object,
  }

  constructor() {
    super()

    this.handleSubmit = this.handleSubmit.bind(this)
    this.handleChange = this.handleChange.bind(this)
  }

  handleChange(evt) {
    const { id, value } = evt.target
    this.setState({
      [id]: value,
    })
  }

  handleSubmit(evt) {
    evt.preventDefault()
    evt.stopPropagation()
    const { history } = this.props
    const { loginEmail, loginPassword } = this.state
    history.push('/')
  }

  render() {
    return (
      <div className={styles.wrapper}>
        <form className={styles.form}>
          <h1>Login</h1>
          <h2>Email</h2>
          <Input
            id="loginEmail"
            onChange={this.handleChange}
            placeholder="Your email"
            type="email"
          />
          <h2>Password</h2>
          <Input
            id="loginPassword"
            onChange={this.handleChange}
            placeholder="Your password"
            type="password"
          />
          <Button
            onClick={this.handleSubmit}
            size="normal"
            uppercase
          >
            login
          </Button>
          <p>Haven&#39;t you got an account?</p>
          <Link to="/register">Sign up.</Link>
        </form>
      </div>
    )
  }
}

export default Login
