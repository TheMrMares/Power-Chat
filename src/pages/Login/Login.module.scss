@import '../../css/config.scss';
@import '../../css/mixins.scss';

.wrapper {
  width: 100%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: $spacing-unit * 1;
}
.form {
  background: $primary-normal;
  padding: $spacing-unit;
  border-radius: $spacing-radius;
  display: flex;
  flex-direction: column;
  flex: 1 1 auto;
  padding: $spacing-unit*2;
  align-items: center;
  border-left: 6px solid;
  border-right: 6px solid;
  border-color: $primary-light;
  min-height: 100%;
  justify-content: center;
  @include respond-to(small) {
    flex: none;
    min-width: 45%;
  }
  @include respond-to(medium) {
    margin: 0;
    min-width: 35%;
  }
  @include respond-to(large) {
    min-width: 25%;
  }
}
