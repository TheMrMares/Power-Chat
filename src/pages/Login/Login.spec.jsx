import React from 'react'
import { shallow } from 'enzyme'
import { Login } from './Login'

const initPage = overrides => {
  const wrapper = shallow(
    <Login
      {...overrides}
    />
  )
  return { wrapper }
}

describe('Login', () => {
  it('should render as expected', () => {
    const { wrapper } = initPage()
    expect(wrapper).toMatchSnapshot()
  })
})
