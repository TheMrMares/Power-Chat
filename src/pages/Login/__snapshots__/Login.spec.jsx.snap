// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render as expected 1`] = `
<div>
  <form>
    <h1>
      Login
    </h1>
    <h2>
      Email
    </h2>
    <Input
      id="loginEmail"
      onChange={[Function]}
      placeholder="Your email"
      type="email"
    />
    <h2>
      Password
    </h2>
    <Input
      id="loginPassword"
      onChange={[Function]}
      placeholder="Your password"
      type="password"
    />
    <Button
      onClick={[Function]}
      size="normal"
      uppercase={true}
    >
      login
    </Button>
    <p>
      Haven't you got an account?
    </p>
    <Link
      replace={false}
      to="/register"
    >
      Sign up.
    </Link>
  </form>
</div>
`;
